Fix Keyboard Accessibility / Mouse Control Issues

dhruvisompura opened the pull request:
Addresses #6451 

This PR resolves the keyboard and mouse control issues with console tab list. Incorrect aria attributes and roving tab index implementations have been fixed. Users should now be able to:
* Use arrow keys (and home/end) to navigate between console tabs 
  * Home/End on Mac are: Fn+ Left Arrow and Fn + Right Arrow
* Pressing Enter when focused on a console tab triggers action to rename session
  * This behavior mirrors the Terminal pane
* Selecting a console tab with the mouse now displays focus indicator
* Keyboard shortcuts to copy/paste/cut/select-all work when renaming a session
* Double clicking or selecting text via the mouse work when renaming a session

A few other housekeeping/polish items were included as well:
* Consolidate duplicate css declaration
* Fix trash icon alignment to match Terminals
* Tweak trash button border radius to match Terminals
* Only show delete button when there is space in console tab

### Release Notes

#### New Features

- N/A

#### Bug Fixes

- Fix keyboard accessibility and mouse control issues in console tab list


### QA Notes
- Home/End on Mac are: Fn+ Left Arrow and Fn + Right Arrow
- When renaming a session, if the user leaves the input field empty, the name of the session does not change.

@:console @:sessions @:win @:web

dhruvisompura committed:
Fix aria attributes

dhruvisompura committed:
Only show delete session button when space allows in console tab

dhruvisompura committed:
Fix trash icon alignment to match Terminals

dhruvisompura committed:
Fix border radius mismatch for delete button

dhruvisompura committed:
Consolidate duplicate css declaration

dhruvisompura committed:
Remove unapplied styles

dhruvisompura committed:
Update error message for console deletion failures

dhruvisompura committed:
Add console tab list to roving tab index

dhruvisompura committed:
Add keyboard navigation to console tab list

dhruvisompura committed:
Polish keyboard and mouse functionality

dhruvisompura committed:
Merge branch 'main' into bugfix/multi-console-keyboard-interactions

Signed-off-by: Dhruvi Sompura <dhruvi.sompura@posit.co>

softwarenerd reviewed (dismissed):
There's one comment on the `tabs-container` `border-top`. It should be changed back to `1px`.

samclark2015 reviewed (dismissed):
Tested on web & desktop, looks great w.r.t. the issue scope!
One issue on web only: cut (`cmd+X`) copies the text, but does not delete from the input (traditional cut).

A couple of thoughts for future accessibility improvements:

- Is there an accessible way to rename? Perhaps `space` when a tab is highlighted?
- When a tab is highlighted, no way to "select" and return focus to console input.

dhruvisompura committed:
Fix tab container border width

dhruvisompura commented:
> Tested on web & desktop, looks great w.r.t. the issue scope! One issue on web only: cut (`cmd+X`) copies the text, but does not delete from the input (traditional cut).

Thanks for catching this! I'll see if I can get that fixed.

> * Is there an accessible way to rename? Perhaps `space` when a tab is highlighted?

I was thinking a user could lean on the command palette, but a faster way to rename a console session would be handy. We can mirror the Terminal pane and have it so pressing `Enter` pulls up the command palette action to rename the session. I can add that in this PR!

> * When a tab is highlighted, no way to "select" and return focus to console input.

I don't think we need to solve for this since the user can just press `Shift+Tab` to navigate back to the console input and `Option+Tab` to navigate back to the console tab list.

dhruvisompura commented on src/vs/workbench/contrib/positronConsole/browser/components/consoleTabList.css:
```diff
@@ -4,16 +4,22 @@
 
 .tabs-container {
-	border-top: 1px solid var(--positronActionBar-border);
+	border-top: 0.5px solid var(--positronActionBar-border);
```
Thanks for catching that! I guess I selected the wrong item during the merge conflict resolution. I still thought this was supposed to be `0.5px`. I'll change it back :)

softwarenerd replied:
YAY!

samclark2015 commented:
I did try shift-tabbing back to console, but the focus seemed to go into the void once it got to where I'd expect the console input to take focus. 🤔

dhruvisompura commented:
> I did try shift-tabbing back to console, but the focus seemed to go into the void once it got to where I'd expect the console input to take focus. 🤔

Ugh ya, I think that happens sometimes - I'm not sure where the focus disappears to but if you press `Tab` or `Shift+Tab` a couple times, it eventually gets back to the console tab list or console input. It's on my radar to look into!

dhruvisompura committed:
Fix cut operation for web

dhruvisompura committed:
Rename session via Enter key when tab is focused

dhruvisompura committed:
Focus tab element on click

dhruvisompura committed:
Ensure console session can be deleted via keyboard

dhruvisompura committed:
Rename functions for easier readability

dhruvisompura commented:
@samclark2015 I updated the PR to include the suggestion you had and fixed the issues on the web version. Give it a go and let me know if you run into any other issues.

dhruvisompura committed:
Fix accessor error when running rename command

samclark2015 reviewed (approved):
LGTM!
